R/shared-methods.R

```diff
@@ -173,10 +178,17 @@ DataClass <- R6::R6Class(
     #' cleaning steps be kept and output in a list.
     #' @param get Logical, defaults to FALSE. Should the class `get` method be
     #' called (this will download, clean, and process data at initialisation).
+    #' @param process_fns array, additional functions to process the data.
+    #' Users can supply their own functions here which would act on clean data
+    #' and they will be called alongside our default processing functions.
+    #' The default optional function added is `set_negative_values_to_zero`.
+    #' If you want to keep this when supplying your own processing functions
+    #' remember to add it to your list also.
     #' @export
     initialize = function(level = "1", regions,
                           totals = FALSE, localise = TRUE,
-                          verbose = TRUE, steps = FALSE, get = FALSE) {
+                          verbose = TRUE, steps = FALSE, get = FALSE,
+                          process_fns = c(set_negative_values_to_zero)) {
```

Comment:
needs to be robust to this being empty which I think currently not?

Reply:
(needs test to check)

Reply:
One step ahead this time [lines 21-25](https://github.com/epiforecasts/covidregionaldata/blob/extend-data-processing/tests/testthat/test-processing.R)